cumulusci/core/tests/test_config.py

```diff
@@ -210,20 +248,328 @@ def get_service(self, name):
+                'root': '.',
+                'url': 'https://github.com/SalesforceFoundation/CumulusCI-Test.git',
+            }, result)
+
+    def test_repo_info_missing_env(self):
+        env = {
+            'CUMULUSCI_AUTO_DETECT': '1',
+            'HEROKU_TEST_RUN_ID': 'TEST1',
+            'HEROKU_TEST_RUN_BRANCH': 'master',
+            'HEROKU_TEST_RUN_COMMIT_VERSION': 'HEAD',
+            'CUMULUSCI_REPO_BRANCH': 'feature/test',
+            'CUMULUSCI_REPO_COMMIT': 'HEAD~1',
+            'CUMULUSCI_REPO_ROOT': '.',
+        }
+        config = BaseProjectConfig(BaseGlobalConfig())
+        with mock.patch.dict(os.environ, env):
+            with self.assertRaises(ConfigError):
+                config.repo_info
+
+    def test_repo_root_from_env(self):
+        config = BaseProjectConfig(BaseGlobalConfig())
+        config._repo_info = {
+            'root': '.',
+        }
+        self.assertEqual('.', config.repo_root)
+
+    def test_repo_name_from_repo_info(self):
+        config = BaseProjectConfig(BaseGlobalConfig())
+        config._repo_info = {
+            'name': 'CumulusCI',
+        }
+        self.assertEqual('CumulusCI', config.repo_name)
+
+    def test_repo_name_no_repo_root(self):
+        config = BaseProjectConfig(BaseGlobalConfig())
+        with temporary_dir() as d:
+            cwd = os.getcwd()
+            os.chdir(d)
+            self.assertIsNone(config.repo_name)
+            os.chdir(cwd)
+
+    def test_repo_name_from_git(self):
+        config = BaseProjectConfig(BaseGlobalConfig())
+        self.assertEqual('CumulusCI', config.repo_name)
+
+    def test_repo_url_from_repo_info(self):
+        config = BaseProjectConfig(BaseGlobalConfig())
+        config._repo_info = {
+            'url': 'https://github.com/SalesforceFoundation/CumulusCI',
+        }
+        self.assertEqual(
+            'https://github.com/SalesforceFoundation/CumulusCI', config.repo_url)
+
+    def test_repo_url_no_repo_root(self):
+        config = BaseProjectConfig(BaseGlobalConfig())
+        with temporary_dir() as d:
+            cwd = os.getcwd()
+            os.chdir(d)
+            self.assertIsNone(config.repo_url)
+            os.chdir(cwd)
+
+    def test_repo_url_from_git(self):
+        config = BaseProjectConfig(BaseGlobalConfig())
+        self.assertIn('SalesforceFoundation/CumulusCI', config.repo_url)
+
+    def test_repo_owner_from_repo_info(self):
+        config = BaseProjectConfig(BaseGlobalConfig())
+        config._repo_info = {
+            'owner': 'SalesforceFoundation',
+        }
+        self.assertEqual(
+            'SalesforceFoundation', config.repo_owner)
+
+    def test_repo_owner_no_repo_root(self):
+        config = BaseProjectConfig(BaseGlobalConfig())
+        with temporary_dir() as d:
+            cwd = os.getcwd()
+            os.chdir(d)
+            self.assertIsNone(config.repo_owner)
+            os.chdir(cwd)
+
+    def test_repo_branch_from_repo_info(self):
```

Comment:
this is repetitive that it feels like maybe we should do `test_from_repo_info` `test_from_git` `test_no_repo_root` that has an assert for each property instead of doing the setup code over and over again.

Reply:
That would have been a good idea, but given that I've already written it this way, I'm not sure it's worth the effort to refactor? This way has the small benefit that a failure on one of the properties doesn't keep it from testing the others.

Reply:
no prob.